src/sagemaker/jumpstart/cache.py

```diff
@@ -338,29 +340,33 @@ def _retrieval_function(
             return JumpStartCachedContentValue(
                 formatted_content=model_specs
             )
-        if data_type == HubDataType.MODEL:
+        if data_type == HubContentType.MODEL:
             hub_name, region, model_name, model_version = utils.extract_info_from_hub_content_arn(
                 id_info
             )
-            hub = CuratedHub(hub_name=hub_name, region=region)
-            hub_content = hub.describe_model(model_name=model_name, model_version=model_version)
+            hub_model_description: HubContentDescription = hub_utils.describe_model(
+                hub_name=hub_name,
+                region=region,
+                model_name=model_name,
+                model_version=model_version
+            )
+            model_specs = JumpStartModelSpecs(hub_model_description, is_hub_content=True)
             utils.emit_logs_based_on_model_specs(
-                hub_content.content_document,
+                model_specs,
                 self.get_region(),
                 self._s3_client
             )
-            model_specs = JumpStartModelSpecs(hub_content.content_document, is_hub_content=True)
+            # TODO: Parse HubContentDescription
```

Comment:
We'll actually parse `hub_model_description.hub_content_document` inside of `JumpStartModelSpecs`. You can see I stubbed out the code for `from_content_document` in that data class